api/v1alpha1/managedocs_types.go

```diff
@@ -38,9 +38,27 @@ type ManagedOCSSpec struct {
 	ReconcileStrategy ReconcileStrategy `json:"reconcileStrategy,omitempty"`
 }
 
+type ComponentStatusMsg string
+
+const (
+	ComponentReady   ComponentStatusMsg = "Ready"
+	ComponentPending ComponentStatusMsg = "Pending"
+	ComponentNone    ComponentStatusMsg = "None"
+	ComponentUnknown ComponentStatusMsg = "Unknown"
+)
+
+type ComponentStatus struct {
+	Status ComponentStatusMsg `json:"status"`
```

Comment:
I believe a more accurate name to this field would be `State` see, 
https://softwareengineering.stackexchange.com/questions/219351/state-or-status-when-should-a-variable-name-contain-the-word-state-and-w#:~:text=status%20is%20used%20to%20describe,(e.g.%20pending%2Fdispatched).

Reply:
"a choice needs to be made involving the perverted intricacies of English language"- what an epithet! ok `State` it is